src/components/Accordion/Accordion.tsx

```diff
@@ -225,15 +225,25 @@ export const Accordion: ChakraComponent<
     );
 
     const handleKeyDown = (e) => {
-      // If the 'esc' key is pressed,
-      if (e.keyCode === 27) {
-        const focusedPanelIndex = Number(e.target.dataset.index);
-        // collapse the currently focused panel.
-        // (Nothing will happen if the currently
-        // focused panel is already collapsed.)
-        setExpandedPanels(
-          expandedPanels.filter((i) => i !== focusedPanelIndex)
-        );
+      // If the 'esc' key is pressed, find the panel the
+      // user is focused on or within, and remove it as
+      // an expanded panel. (Nothing will happen if the
+      // panel is already collapsed.)
+      if (e.code === "Escape") {
+        let focusedPanelIndex;
+        if (e.target.dataset.index) {
+          // If the user is focused on an accordion button...
+          focusedPanelIndex = Number(e.target.dataset.index);
+        } else {
+          // If the user is focused on an element within the panel...
+          focusedPanelIndex = Number(
+            e.target.closest("[role='region']").id.split("-").pop()
+          );
+
+          setExpandedPanels(
+            expandedPanels.filter((i) => i !== focusedPanelIndex)
+          );
```

Comment:
This is within the `else` block and it doesn't seem intentional because the esc key should still be functional when the user is focused on the accordion button.

Reply:
Yes! Good catch. Thank you.